### PR TITLE
reports: document riskdesc field

### PR DESCRIPTION
**Pull request by sidejackthenativity**
Added documentation to template file. This is to clarify ambiguity of output. Specifically, level of risk and level of confidence.

Fix zaproxy/zaproxy#7445

### CODE REVIEW

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

File: addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-json.html
Comment (kingthorin) [inline addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-json.html:29]: I'd suggest documenting this between the `H3` and `pre` tags. Maybe something more like:

```html
<H4>riskdesc</H4>
riskdesc - Is a combination identifier, showing Risk followed by Confidence (in brackets). For example <code>High (Medium)</code> would indicate a High risk issue with Medium confidence.
```

Comment (kingthorin) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-json.html:29]: @thc202 @psiinon @ricekot thoughts?

Comment (thc202) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-json.html:29]: That seems a better place.

File: addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-json.html
Comment (kingthorin) [inline addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-json.html:10]:
```suggestion
    <H4>About riskdesc</H4>
```

**Comment by kingthorin** [comment]
@sidejackthenativity are you going to address the DCO requirement?

If you're worried about it you can just agree to https://developercertificate.org/ in a comment and we can tweak things on your behalf.

**Comment by sidejackthenativity** [comment]
@kingthorin Hi. No, I am happy to sign it. I am still trying to get accustomed to github at this level. I tried to sign my consequent commits but I guess it didn't override the initial one. I will spend a little time and figure it out. Sorry, bare with me. :)

**Comment by kingthorin** [comment]
No problem, let us know if you get stuck.

**Comment by sidejackthenativity** [comment]
Hi there. I don't know how to address the first commit lack of signature. I tried rebasing but I think it is best to ask for help at this point. Any tips on how to do this would be appreciated.

**Comment by sidejackthenativity** [comment]
I think I figured it out. Github is surprisingly helpful.

**Comment by kingthorin** [comment]
Seems really close, there’s some unrelated commits now. I can put together some instructions to address that in the morning.

Thanks for sticking with it.

**Comment by kingthorin** [comment]
You'll want to do an interactive rebase on the 11 commits you've included:

`git rebase -i HEAD~11`

> It'll drop you into an editor, for me it's vi. They'll all be listed starting with pick. So change it like the following (`d` for drop, leave the first one you submitted as `pick`, then `f` for fixup on the others):

d 33e30ce12 network: use custom executor to process messages
d bf4ddd321 network: update dependencies
d a363d9c24 spelling: addOns/accessControl
d ba6e98a56 spelling: addOns/alertFilters
d e77ba838e spelling: addOns/amf
d e1923d05f Automation: Warn if unexpected element under activeScan job
pick e07537c5a adding documentation to template file
f 99be3a9fd Added documentation to template file. This is to clarify ambiguity of output. Specifically, level of risk and level of confidence.
f 4e234c50f adding documentation to template file
f 30a4fc55a Added documentation to template file. This is to clarify ambiguity of output. Specifically, level of risk and level of confidence.
f bdf85631b Update addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-json.html

> Use `esc` `:wq` to write and quit vi.

You should see "Successfully rebased and updated refs/heads/doc-changes."

You can then check:

`git log -1`

Should display something like:

```console
Author: sidejackthenativity <sidejackthenativity@gmail.com>
Date:   Wed Dec 21 09:13:17 2022 -0500

    adding documentation to template file

    Signed-off-by: sidejackthenativity <sidejackthenativity@gmail.com>
```

`git push --force-with-lease`

You should now see your changes here with a single signed off commit.

**Comment by sidejackthenativity** [comment]
Thanks for spending the time to go over that with me. I really appreciate your time and patience. If this change is accepted I will begin to work on the other documents after the holidays.

**Comment by kingthorin** [comment]
@sidejackthenativity no problem. We've all been there at some point. I'm good with the wording. @psiinon @thc202 @ricekot could we get a second opinion?

**Comment by kingthorin** [comment]
Looks like the changes could use a spotlessApply.

```bash
// This first one applies the code formatting the project expects

./gradlew :aO:reports:spotlessApply
// This one adds the changed files to the last commit, without editing the commit message, and does apply sign-off (though shouldn't be necessary because it already has it (can't hurt to be careful)

git commit -a --amend --no-edit --signoff 
```

**Comment by thc202** [comment]
Sounds good to me.

**Comment by sidejackthenativity** [comment]
> Looks like the changes could use a spotlessApply.
> 
> ```shell
> // This first one applies the code formatting the project expects
> 
> ./gradlew :aO:reports:spotlessApply
> 
> // This one adds the changed files to the last commit, without editing the commit message, and 
> // does apply sign-off (though shouldn't be necessary because it already has it (can't hurt to be careful)
> 
> git commit -a --amend --no-edit --signoff 
> ```

I sent those commands. And i had to do some merging with my repo. Hopefully, that did the trick.

**Comment by kingthorin** [comment]
That doesn't seem right. What's telling you, you need to merge?

Can you do a `git status` between the commands and let us know what it says? Because it seems like the changes didn't apply.

**Comment by sidejackthenativity** [comment]
This is what it says:
```
(base) Ryans-Air-2:zap-extensions ryanthompson$ git status
On branch doc-changes
Your branch and 'origin/doc-changes' have diverged,
and have 2 and 32 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

nothing to commit, working tree clean
```

**Comment by kingthorin** [comment]
That was after a spotlessApply? Or that’s just what it’s saying now (without having done anything)?

**Comment by sidejackthenativity** [comment]
That was after the two commands. Then I did a git pull. Should I do the commands over?

```
(base) Ryans-Air-2:zap-extensions ryanthompson$ ./gradlew :aO:reports:spotlessApply
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :buildSrc:jar
:jar: No valid plugin descriptors were found in META-INF/gradle-plugins

> Configure project :addOns:browserView
Project :addOns:browserView => no module-info.java found

BUILD SUCCESSFUL in 53s
23 actionable tasks: 14 executed, 6 from cache, 3 up-to-date
(base) Ryans-Air-2:zap-extensions ryanthompson$ git commit -a --amend --no-edit --signoff 
[doc-changes bece676f7] adding documentation to template file
 Date: Wed Dec 21 09:13:17 2022 -0500
 1 file changed, 5 insertions(+), 1 deletion(-)
(base) Ryans-Air-2:zap-extensions ryanthompson$ git status
On branch doc-changes
Your branch and 'origin/doc-changes' have diverged,
and have 1 and 1 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

nothing to commit, working tree clean
(base) Ryans-Air-2:zap-extensions ryanthompson$ git pull```

**Comment by kingthorin** [comment]
Okay, my fault, I took a step for granted. Don't `git pull` at the end. Instead force your changes to the remote. So:

```text
git reset --hard 43e5d5c8a92d3e3bb8f8396802000458a05ee6f6
./gradlew :aO:reports:spotlessApply
git commit -a --amend --no-edit --signoff
git push --force-with-lease
```

Reset to the specific commit with just your changes. Run the spotlessApply. Do an amending commit adding the changes file(s) without editing the commit message and sign it off for good measure. Force push it to your remote, which will update this PR.

**Comment by kingthorin** [comment]
Yay build passed. Okay so that same change can be made in the other templates too 🙂

**Comment by sidejackthenativity** [comment]
@kingthorin Thanks for your help! It seemed like a xmas miracle! LOL Enjoy yourself. I will make the additional changes late next week!

**Comment by kingthorin** [comment]
Thanks for sticking with everything. No rush, enjoy your holidays 🎄

**Comment by thc202** [comment]
The changelog should be updated.

**Comment by kingthorin** [comment]
@sidejackthenativity do you still plan to tackle the rest or should one of us finish this off?

**Comment by sidejackthenativity** [comment]
@kingthorin No, I am going to finish it. Sorry. I have been busy. I wanted to confirm the process:
1. Sync my fork with the latest build
2. Pull request to my local rep
3. make changes
4. push SIGNED changes
5. pray?

**Comment by kingthorin** [comment]
1. Fetch upstream.
2. Rebase your local.
3. Make changes and do a signed-off amending commit.
4. Push changes.
5. If all went well hopefully this PR is updated appropriately.

**Comment by sidejackthenativity** [comment]
Thank you. My goal is to complete this by Sunday evening.

**Comment by thc202** [comment]
Note that in step 4 you'll probably have to use `-f`.

**Comment by kingthorin** [comment]
Good point, tweaked.

**Comment by sidejackthenativity** [comment]
@kingthorin I appreciate the thoroughness there!

**Comment by sidejackthenativity** [comment]
Should step 4 have the force or step 3?

**Comment by kingthorin** [comment]
4, sorry. Fixing now.

**Comment by kingthorin** [comment]
Thanks

**Comment by thc202** [comment]
There's a typo in clarify.

**Comment by thc202** [comment]
Those steps look correct.

**Comment by sidejackthenativity** [comment]
I have seemed to have broken something again...

**Comment by kingthorin** [comment]
It needs a spotlessApply again.

**Comment by thc202** [comment]
The changelog still needs to be updated.

**Comment by sidejackthenativity** [comment]
Like this?
```
git fetch upstream/main
git rebase upstream/main
./gradlew :aO:reports:spotlessApply
git commit --amend --signoff 
git push --force
```

Should I do a hard reset again? or not necessary this time?

**Comment by kingthorin** [comment]
For the spotlessApply you only need the bottom three. To address the CHANGELOG.md bit you could do that between spotlessApply and commit (add `-a` on the commit to include newly modified files...the changelog [and no-edit because you don't need to change the commit message]).

```
./gradlew :aO:reports:spotlessApply
// Edit CHANGELOG.md adding a note (suggestion below), save it ;)
git commit -a --amend --no-edit --signoff 
git push --force
```

```md
### Added
- A description of riskdesc fields in the relevant report templates help.
```

**Comment by sidejackthenativity** [comment]
You are referring to the change file: 

zap-extensions/addOns/reports/CHANGELOG.md

**Comment by kingthorin** [comment]
Yup that's the one.

File: addOns/reports/CHANGELOG.md
Comment (thc202) [inline addOns/reports/CHANGELOG.md:8]: Add a newline between the subsections.

File: addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html
Comment (thc202) [inline addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Why is this one different in all cases there's an example?

File: addOns/reports/CHANGELOG.md
Comment (sidejackthenativity) [reply addOns/reports/CHANGELOG.md:8]: Ok.

File: addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html
Comment (sidejackthenativity) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: > Why is this one different in all cases there's an example?

I did that because the Riscdesc in that particular template is very far down the file and it seemed weird to be at the top without more information. I can change back to the others though.

Comment (thc202) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Agreed this one needs to be explained differently than the others (since it's not a named field), how about:
```html
<H4>Header <code>Risk (Confidence)</code></H4>
This header is a combination identifier, showing Risk followed by Confidence. For example <code>High (Medium)</code>, would indicate a High risk issue identified with Medium confidence.
```

Comment (sidejackthenativity) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Yeah, I can do that.

Comment (kingthorin) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Thank you both.

Comment (sidejackthenativity) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: I appreciate both your patience. I am glad I started with something simple, the version control of this is a big learning curve.

Comment (sidejackthenativity) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: So if I have altered 2 files since the last push (Changelog and the template file to accomodate thc202 wording), is it the same commit process?

git rebase
./gradlew :aO:reports:spotlessApply
git commit -a --amend --no-edit --signoff 
git push --force`

Comment (kingthorin) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Shouldn’t need the rebase, bottom 3 steps are good.

Comment (sidejackthenativity) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Thanks. I am getting the following error: 

`$ git commit -a --amend --no-edit --signoff 
fatal: this operation must be run in a work tree`

Shoulld I exlude the --amend?

Comment (kingthorin) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Could you provide the output from `git status`? I don't understand why it thinks you're mid-merge.

Comment (sidejackthenativity) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: sure...

`On branch doc-changes
Your branch is up to date with 'origin/doc-changes'.

All conflicts fixed but you are still merging.
  (use "git commit" to conclude merge)

Changes to be committed:
        modified:   addOns/reports/CHANGELOG.md
        modified:   addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html
        modified:   addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-xml-plus.html

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html
        modified:   addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-xml-plus.html

`

Comment (kingthorin) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Did you actually start with a rebase?

Comment (kingthorin) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: That'd explain why it thinks you're merging.

Comment (sidejackthenativity) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: I did a git pull but not a rebase

Comment (sidejackthenativity) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Maybe my thoughts on how this works is not correct but I assume I need to grab the current version of the repo before I do anything?

Comment (kingthorin) [reply addOns/reports/src/main/javahelp/org/zaproxy/addon/reports/resources/help/contents/report-traditional-markdown.html:14]: Okay pull is the same as fetch and merge. That's why.

You're better off doing a fetch and rebase, that way you don't end up with unrelated commits in your PR.

For the time being I'd suggest `git merge --abort` and go ahead with what you're trying to do.

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
@thc202 / @psiinon if you’re happy with this I can flatten it or it can be squashed on merge. Just let me know.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by kingthorin** [comment]
Thanks everyone!

**Comment by sidejackthenativity** [comment]
Thank you both!

**Comment by kingthorin** [comment]
@sidejackthenativity 

Just to kinda come full circle on this, here's a summary of things when you want to tackle an issue.

1. Make sure your fork is up-to-date. You can just use "Sync fork" in the GitHub webui.
2. Make sure your local is up-to-date.
```bash
git checkout main
git fetch upstream/main
git merge
```
3. Create a new branch for the think you're going to work on/change: `git checkout -b some-new-branch` (Don't pull or fetch & merge on your feature branch).
4. Make your changes/additions.
5. You can check what you've changed and what is/isn't tracked `git status`.
6. Commit your work (In general: `git commit -a -m "Cool thing I fixed" -m "Updated CHANGELOG. Added lmnop. Tweaked tests cases" --signoff`. If you have new/untracked files you might need to specifically `git add /path/to/file` before committing).
7. Push it to your fork `git push -u origin some-new-branch`.
8. Goto the GitHub web interface and open your PR.

If you need to update based on upstream changes and address conflicts:
```bash
git checkout main
git fetch upstream/main
git merge
git checkout some-new-branch
git rebase upstream/main
# Address CONFLICTS and save files
git add path/to/some/changed/file
git rebase --continue
git push --force
```

If you need fix things in your PR:
```bash
git checkout some-new-branch
# Make changes/corrections and save files (if you need to add or move files here you may have untracked files and need to git add as outlined above
git commit -a --amend --no-edit --signoff
git push --force
```

I'm sure that looks like a lot but you won't often need to address conflicts or update based on upstream. It just depends how long things take and what else is going on.

**Comment by kingthorin** [comment]
@sidejackthenativity how would you like to be credited? (name, handle etc) 

Have a look at the current entries: https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team and let us know.

**Pull request merged**
